utils/src/main/java/software/amazon/awssdk/utils/cache/NonBlocking.java

```diff
@@ -37,80 +46,204 @@
 public class NonBlocking implements CachedSupplier.PrefetchStrategy {
     private static final Logger log = Logger.loggerFor(NonBlocking.class);
 
+    /**
+     * The maximum number of concurrent refreshes before we start logging warnings and skipping refreshes.
+     */
+    private static final int MAX_CONCURRENT_REFRESHES = 100;
+
+    /**
+     * The {@link Random} instance used for calculating jitter of the background prefetches.
+     */
+    private static final Random JITTER_RANDOM = new Random();
```

Comment:
Perhaps consider ThreadLocalRandom for better performance

Reply:
We should keep an eye out on our benchmarks and fix this if it appears, but it is used infrequently enough (once per credential refresh) that I'd be surprised if any parallel calls are contending over its lock.